sdk/python/kfp/compiler/compiler_utils.py

```diff
@@ -274,7 +274,13 @@ def get_inputs_for_all_groups(
                             channels_to_add.pop()
                             if not channels_to_add:
                                 break
-
+            elif isinstance(channel, for_loop.Collected):
+                # Don't create compiler-injected inputs for fan-in.
+                # This is data passing in the opposite direction: instead of
+                # passing inputs from a parent task group to a child task
+                # group/task (what this function handles), instead surface
+                # outputs from inner tasks to one or more parent groups
```

Comment:
I couldn't think through this clearly. If we bypassing the injection, would the following case work? 
```python
with dsl.ParallelFor([1, 2, 3]) as f:
    t = double(f)
with dsl.Condition(...):
    add(nums=dsl.Collected(t.output))
```
What would be the "punching-hole" input shown in the condition DAG component spec?

Reply:
Good catch. This is unnecessary logic. This was holdover code from a previous iteration of my work on this PR. The "hole punching" is done via the `if channel.task_name` condition.

I've removed this. [This test case](https://github.com/kubeflow/pipelines/blob/f51729ef1665eeafe35aa1fbd92529a9dafca833/sdk/python/test_data/pipelines/parallelfor_fan_in/parameters_with_conditional_consumer.py) verifies that compilation works correctly for the case you are describing.